rep-0008.rst

```diff
@@ -22,27 +25,148 @@ Please follow Python's PEP 8 [1]_ for guidelines on formatting, naming, etc...
 
-roslib is the only ROS Python package you can automatically assume to
+With ``catkin``
+---------------
+
+The loader method of accessing Python code in ROS is obsolete if you are using
+``catkin``.
+When using ``catkin``, the Python code for all packages will be on the
+``PYTHONPATH`` after sourcing the ``setup`` file for your workspace.
+
+With ``rosbuild``
+-----------------
+
+``roslib`` is the only ROS Python package you can automatically assume to
 be importable other than your local files. If your Python package has
 dependencies, you must include the following header at the top::
 
     import roslib
     roslib.load_manifest('your_package_name')
 
-roslib.load_manifest looks at your manifest.xml and places your
-dependencies on your Python path. DO NOT use multiple load_manifest
+``roslib.load_manifest`` looks at your ``manifest.xml`` and places your
+dependencies on your Python path. DO NOT use multiple ``load_manifest``
 calls. If you need multiple calls, it's probably because you're
 missing the correct dependencies in your manifest.
 
 
-Package/Module Names (__init__.py files)
-========================================
+Package/Module Names (``__init__.py`` files)
+============================================
+
+With ``catkin``
+---------------
+
+When using ``catkin`` you can organize your Python code in any way that you like, but you must inform Python's ``setuptools`` where your code is located by creating a ``setup.py`` file.
+The recommended layout is to put all of your package's importable Python code in the ``src`` folder of your package and in a "Python package" with the same name as your package.
+In this case a "Python package" is a folder which contains a ``__init__.py`` file, indicating that the folder contains additional Python packages and modules.
+
+Here is the recommended layout of an example ``catkin`` package called ``my_package`` which contains Python code::
+
+    my_package
+    ├── CMakeLists.txt
+    ├── package.xml
+    ├── setup.py
+    └── src
+        └── my_package
+            ├── __init__.py
+            ├── a_nested_python_package
+            │   └── __init__.py
+            └── a_python_module.py
+
+In this example there is a root Python package in the ``src`` folder called ``my_package``, which follows the recommendation.
+This root Python package then contains a Python module called ``a_python_module`` and a nested Python package called ``a_nested_python_package``.
+
+In this example the ``setup.py`` will use the convenience function ``generate_distutils_setup``, which is provided by ``catkin``, in order to extract information like the version and description from the
+package's manifest file, i.e. the ``package.xml``:
+
+.. code-block:: python
+
+    from distutils.core import setup
+    from catkin_pkg.python_setup import generate_distutils_setup
+
+    d = generate_distutils_setup(
+        packages=['my_package', 'my_package.a_nested_python_package'],
+        package_dir={'': 'src'},
```

Comment:
But that would lead to more redundant information I think. I don't understand how that would be simpler. Please provide a complete example to demonstrate what you mean. Also the above is the first recommendation for "custom" layouts of source in the Python docs: https://docs.python.org/2/distutils/setupscript.html#listing-whole-packages